docs/adr/0003-use-of-devcontainer-directory.md

```diff
@@ -0,0 +1,55 @@
+# 3. Use of the .devcontainer directory in the research-template repo
+
+Date: 2024-06-04
+
+## Status
+
+Accepted
+
+## Context
```

Comment:
I'm going to read back what I think I've learnt from the "Context section:

* `devcontainer.json` contains configuration for a dev container
* `devcontainer.json` references a Docker image
* whilst some configuration is pushed upstream to the Docker image, some must stay downstream in `devcontainer.json` (e.g. setting R/RStudio's working directory)
* `devcontainer.json` depends on supporting files
* `.devcontainer` is a standard location for `devcontainer.json` and supporting files
* research-template must have `.devcontainer`
* each research repo must have `.devcontainer`
* it's important to establish who can reasonably be expected to update `.devcontainer` in each case
* it's important that each research repo's `.devcontainer` doesn't drift from research-template's `.devcontainer`
* but there be reasons why a research repo's `.devcontainer` does drift

Reply:
Yep, that's correct